File: content/be/mlops/source-code/dvc/experiment-management-python/src/experiment.py
#!/usr/bin/env python3

import argparse
import dvclive


def parse_args():
    parser = argparse.ArgumentParser(description="Run an experiment with specified parameters.")
    parser.add_argument("--a", type=float, default=1.0, help="Parameter a")
    parser.add_argument("--t_max", type=int, default=10, help="Maximum number of time steps")
    args = parser.parse_args()
    return args

def function(t, a):
    """A simple function to simulate some behavior."""
    return a * t**2

def main():
    args = parse_args()
    with dvclive.Live(report='html') as live:
        live.log_param("a", args.a)
        live.log_param("t_max", args.t_max)
        for t in range(args.t_max + 1):
            result = function(t, args.a)
            live.log_metric("result", result)
            live.next_step()

if __name__ == "__main__":
    main()

File: content/be/mlops/source-code/dvc/experiment-management-r/src/experiment.R
#!/usr/bin/env Rscript

library(optparse)
library(reticulate)

option_list <- list(
  make_option(c("--a"), type = "double", default = 1.0, help = "Parameter a"),
  make_option(c("--t_max"), type = "integer", default = 10, help = "Maximum number of time steps")
)
parser <- OptionParser(option_list = option_list, description = "Run an experiment with specified parameters.")
opts <- parse_args(parser)

py_module_available("dvclive") || stop("Python module 'dvclive' is not available")
dvclive <- import("dvclive")

live <- dvclive$Live(report = "html")

compute <- function(t, a) {
  a * t^2
}

live$log_param("a", opts$a)
live$log_param("t_max", opts$t_max)

for (t in seq(0, opts$t_max)) {
  result <- compute(t, opts$a)
  live$log_metric("result", result)
  live$next_step()
}

live$end()